runtime-modules/forum/src/lib.rs

```diff
@@ -790,6 +787,38 @@ decl_module! {
             Ok(())
         }
 
+        /// Update category
+        fn update_thread_archival_status(origin, actor: PrivilegedActor<T>, category_id: T::CategoryId, thread_id: T::ThreadId, new_archival_status: bool) -> dispatch::Result {
+            // Ensure data migration is done
+            Self::ensure_data_migration_done()?;
+
+            // Ensure actor can update category
+            let (category, thread) = Self::ensure_can_update_thread_archival_status(origin, &actor, &category_id, &thread_id)?;
```

Comment:
This name `ensure_can_update_thread_archival_status` seems to suggest that its conclusive single guarantor that the actor can update the archival state of a thread, however, there are all the remaining checks from line 798 which seem to be checking additional things relevant to that assessment. It seems that either those checks really should be part of this one check, or, the name of this check really is incorrect, as it providing some other check.

Reply:
The big chunk of code you mentioned is removed now.